python/lsst/ts/aos/utils/m1m3/m1m3_aav.py

```diff
@@ -0,0 +1,761 @@
+        end_time: Time
+            Time to look till.
+        """
+
+        # find last state before start_time
+        logging.info(f"Retrieve last detailedState before {start_time.isot}.")
+        query = (
+            "SELECT detailedState "
+            'FROM "efd"."autogen"."lsst.sal.MTM1M3.logevent_detailedState" '
+            f"WHERE time <= '{start_time.isot}+00:00' ORDER BY time DESC LIMIT 1"
+        )
+        ret = await self.client.influx_client.query(query)
+        detailed_states = await self.client.select_time_series(
+            "lsst.sal.MTM1M3.logevent_detailedState",
+            "detailedState",
+            start_time,
+            end_time,
+        )
+        self.detailed_states = pd.concat([ret, detailed_states])
+        self.detailed_states.set_index(
+            pd.DatetimeIndex(self.detailed_states.index), inplace=True
+        )
+
+    def was_raised(self, start: pd.Timestamp, end: pd.Timestamp) -> bool:
+        """Return whenever mirror ws raised (in either ACTIVE or
+        ACTIVEENGINEERING states) from start till end parameters.
+
+        Parameters
+        ----------
+        start : Time
+            Starting time of interval to query for mirror state.
+        end : Time
+            End time of interval to query for mirror state.
+
+        Returns
+        -------
+        raised : bool
+            True if mirror was raised (in ACTIVE or ACTIVEENGINEERING
+            detailedState) from start till end.
+        """
+        assert self.detailed_states is not None
+        last_state = self.detailed_states[
+            self.detailed_states.index < pd.to_datetime(start, utc=True)
+        ]["detailedState"].iloc[-1]
+        if DetailedState(last_state) not in (
+            DetailedState.ACTIVE,
+            DetailedState.ACTIVEENGINEERING,
+        ):
+            return False
+        states = self.detailed_states[
+            (Time(self.detailed_states.index) >= start)
+            & (Time(self.detailed_states.index) <= end)
+        ]
+        return (
+            len(
+                states[
+                    ~states.detailedState.isin(
+                        [
+                            DetailedState.ACTIVE,
+                            DetailedState.ACTIVEENGINEERING,
+                        ]
+                    )
+                ].index
+            )
+            == 0
+        )
+
+    async def find_az_el(self, start: Time, end: Time) -> None:
+        """Find times when elevations and azimuth MTMount axis moved. Store
+        elevations and azimuth to elevations and azimuths attributes.
+
+        Parameters
+        ----------
+        start : Time
+            Start time. Azimuths and elevations data will be available from
+            that time.
+        end : Time
+            End time. Azimuths and elevations data will be available till that
+            time.
+        """
+
+        async def find_axis(axis: str) -> pd.DataFrame:
```

Comment:
describe what it does in a comment, please

Reply:
Documented.